Server/Ebenezer/User.cpp

```diff
@@ -12070,6 +12074,77 @@ bool CUser::CheckClass(short class1, short class2, short class3, short class4, s
 	return false;
 }
 
+bool CUser::CheckPromotionEligible()
+{
+	CNpc* npc = m_pMain->m_NpcMap.GetData(m_sEventNid);
+	if (npc == nullptr)
+		return false;
+
+	short tempClass = -1;
+
+
+	if (CheckClass(CLASS_KA_GUARDIAN, CLASS_KA_PENETRATOR, CLASS_KA_NECROMANCER, CLASS_KA_DARKPRIEST) 
+		|| CheckClass(CLASS_EL_PROTECTOR, CLASS_EL_ASSASIN, CLASS_EL_ENCHANTER, CLASS_EL_DRUID))
+	{
+		if (m_pUserData->m_sClass == CLASS_EL_PROTECTOR || m_pUserData->m_sClass == CLASS_KA_GUARDIAN)
+		{
+			SendSay(-1, -1, 6006, -1, -1, -1, -1, -1, -1, -1);
```

Comment:
Ideally we should take out the trailing -1s to let it use the defaults, and perhaps add a comment for what 6006 says in the client (likewise for the other cases).

Reply:
Quest_Talk_us.tbl

Reply:
that was much to obvious xD